sql/opt_range.cc

```diff
@@ -8029,8 +8031,8 @@ and_all_keys(RANGE_OPT_PARAM *param, SEL_ARG *key1, SEL_ARG *key2,
 
   if (key1->elements != 1)
```

Comment:
What is this condition for? Previously it looked like we wanted to avoid invocation of `key2->increment_use_count(0)` but now we skipping `key2->increment_use_count(1)`?